### PR TITLE
establish Playwright E2E testing infrastructure with configuration

### PR DESCRIPTION
**Pull request by devvaansh**
## Description

This pull request establishes the foundation for Playwright based end to end (E2E) testing in the web application. It introduces the necessary configuration, a GitHub Actions workflow for CI, and initial test cases for the login functionality, following the Page Object Model (POM) for better maintainability.

## Playwright Configuration:

Added playwright.config.ts to define the test environment, including browser settings, timeouts, and reporting.
Included tsconfig.playwright.json for TypeScript compilation of tests.
## CI/CD Integration:

A new GitHub workflow in .github/workflows/playwright.yml is set up to automatically run Playwright tests on each push and pull request to the main branch, ensuring continuous integration.
## Page Object Model (POM):

Implemented the Page Object Model to create a scalable and maintainable test framework.
playwright/pages/BasePage.ts: A base page with common functionalities.
playwright/pages/login.page.ts: A page object representing the login page, encapsulating its elements and user interactions.
Initial Login Tests (playwright/tests/login.spec.ts):

## Happy Path:
A test case to verify that a user with valid credentials can successfully log in.

## Invalid Paths (9 tests):
Tests to ensure the application correctly handles various invalid login scenarios, such as:
Empty username and password.
Valid username and empty password.
Empty username and valid password.
Invalid username and valid password.
Valid username and invalid password.
Invalid username and invalid password.
and more to cover different edge cases.
## Dependencies:

Added @playwright/test to package.json.
This initial setup paves the way for expanding E2E test coverage across the application, improving overall quality and reliability.

## Screenshots, if any

test going on in web browser-

https://github.com/user-attachments/assets/b008ae87-c87e-48cf-8e09-fbfd4dd51fd1

Screenshot 2025-12-18 at 3 17 06 AM

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [X] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

## Summary by CodeRabbit

* **Tests**
  * Added automated end-to-end testing framework with comprehensive test coverage for login and authentication workflows.
  * Configured test environment with reporting capabilities and continuous integration support.

* **Chores**
  * Updated project configuration files and dependencies to support testing infrastructure.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by devvaansh** [comment]
@alberto-art3ch  This is the Proof of Concept (PoC) for the Playwright infrastructure we discussed in chat. It establishes the base configuration and Page Object Model structure. I am ready to scale this to the Client Module upon your feedback.

File: playwright/tests/login.spec.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    // Attempt login with wrong password
+    await loginPage.login('mifos', 'wrongpassword');
+
+    // Wait for the login attempt to complete (loading indicator disappears)
+    await page.waitForTimeout(3000);
+
+    // Should remain on login page or show error
+    // The form should be enabled again after failed attempt
+    await expect(loginPage.usernameInput).toBeEnabled({ timeout: 10000 });
+  });
```
[the remark this answers is not in the document]
Comment (alberto-art3ch) [reply playwright/tests/login.spec.ts:87]: Please consider this coderrabit suggestion

**Review by alberto-art3ch — CHANGES_REQUESTED** [review]
LGTM only consider that suggestion

**Comment by devvaansh** [comment]
Hi Alberto, thanks for flagging the CI error.

I have just pushed a fix that resolves both issues

File: playwright/pages/login.page.ts [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    return this.page.getByRole('textbox', { name: 'Username' });
+  }
+
+  /**
+   * Get the password input wrapper div.
+   * Uses codegen selector: div.filter({ hasText: 'Password' }).nth(5)
+   */
+  get passwordDivWrapper(): Locator {
+    return this.page.locator('div').filter({ hasText: 'Password' }).nth(5);
+  }
```
[the remark this answers is not in the document]
Comment (devvaansh) [reply playwright/pages/login.page.ts:48]: Acknowledged. I am aware that .nth() selectors are brittle. However, due to the complex DOM structure of the legacy application and the lack of specific test attributes (like data-testid), these Codegen selectors are currently the most reliable way to interact with the wrapper elements.

I am keeping these for this initial PoC to ensure the tests pass. I plan to refactor them into robust locators (or add proper ID attributes to the source code) in a future PR as part of the GSoC roadmap.

**Comment by gkbishnoi07** [comment]
@devvaansh your workflow failed please check it

**Comment by devvaansh** [comment]
@gkbishnoi07 @alberto-art3ch The latest push should now pass all checks. Please let me know if you see any further issues!

**Review by alberto-art3ch — APPROVED** [review]
LGTM
